src/runner/runESLint.js

```diff
@@ -1,7 +1,94 @@
-const { pass, fail, skip } = require('create-jest-runner');
 const { CLIEngine, ESLint } = require('eslint');
 const getESLintOptions = require('../utils/getESLintOptions');
 
+/*
+ * This function exists because there are issues with the `pass`, `skip`, and
+ * `fail` functions from `create-jest-runner`:
+ *
+ * 1. The `pass` and `skip` functions have a bug in our version of
+ *    `create-jest-runner` where calling them will actually pass an `undefined`
+ *    item in `testResults`, which causes the GithubActionsReporter to report an
+ *    empty error message on every file.  This has been resolved in later
+ *    versions of `create-jest-runner`, but upgrading is a breaking change that
+ *    is incompatible with some node versions we support.
+ *
+ * 2. The `fail` function in `create-jest-runner` does not support passing
+ *    multiple failure messages, which makes it impossible to annotate each
+ *    eslint failure.  This has not been resolved, although presumably could be
+ *    worked around by using the underlying `toTestResult` function instead
+ *    (although that function isn't exposed in the public API of that library).
+ *
+ * TODO At some point, we should put a PR in to `create-jest-runner` to resolve
+ * point 2 above, and then should upgrade and remove this function and go back
+ * to using `pass`, `skip`, and `fail` from that library instead.
+ */
+const mkTestResults = ({
+  message,
+  start,
+  end,
+  numFailingTests,
+  numPassingTests,
+  testPath,
+  assertionResults,
+}) => {
+  const startTime = new Date(start).getTime();
+  const endTime = new Date(end).getTime();
+
+  return {
+    failureMessage: message,
+    leaks: false,
+    numFailingTests,
+    numPassingTests,
+    numPendingTests: 0,
+    numTodoTests: 0,
+    openHandles: [],
+    perfStats: {
+      start: startTime,
+      end: endTime,
+      duration: endTime - startTime,
+      slow: false,
+    },
+    skipped: numPassingTests === 0 && numFailingTests === 0,
+    snapshot: {
+      added: 0,
+      fileDeleted: false,
+      matched: 0,
+      unchecked: 0,
+      uncheckedKeys: [],
+      unmatched: 0,
+      updated: 0,
+    },
+    testFilePath: testPath,
+    testResults: assertionResults.map(result => ({
+      duration: endTime - startTime,
+      ancestorTitles: [],
+      failureDetails: [],
+      failureMessages: result.message ? [result.message] : [],
+      fullName: result.fullName,
+      location: result.location,
+      testFilePath: testPath,
+      numPassingAsserts: numPassingTests,
+      status: result.status,
+      title: result.title,
+    })),
+  };
+};
+
+const mkAssertionResults = (testPath, report) =>
+  report[0].messages?.map(reportMessage => ({
```

Comment:
why only `report[0]`? Is there never more than one?

Reply:
That's a good question, there's only ever one, but only because each result represents a file and on [line 186](https://github.com/jest-community/jest-runner-eslint/blob/aa2856c4d4715cfe0727fa63f97484dd32ae6f18/src/runner/runESLint.js#L186) we are passing in just a single file at a time to the eslint api.

I suppose I could change this so that it flat maps across the messages inside arbitrary number of reports just in case we were to ever move to using eslint in batch mode for whatever reason, but there's a lot more that wasn't added in this PR that would have to be changed too, such as [line 198](https://github.com/jest-community/jest-runner-eslint/blob/aa2856c4d4715cfe0727fa63f97484dd32ae6f18/src/runner/runESLint.js#L198), [line 212](https://github.com/jest-community/jest-runner-eslint/blob/aa2856c4d4715cfe0727fa63f97484dd32ae6f18/src/runner/runESLint.js#L212), and [line 239](https://github.com/jest-community/jest-runner-eslint/blob/aa2856c4d4715cfe0727fa63f97484dd32ae6f18/src/runner/runESLint.js#L239), all of which are making the same assumption.  So when I was writing this, it definitely felt icky, but it didn't feel like the right PR to make that change in.

Happy to make that change if you'd prefer though, it's pretty simple.

Reply:
Sounds like a separate PR. Wanna send that one? 🙂